webdriver/tests/bidi/browsing_context/print/margin.py

```diff
@@ -105,14 +109,55 @@ async def test_margin_same_as_page_dimension(
     await bidi_session.browsing_context.navigate(
         context=top_context["context"], url=page, wait="complete"
     )
-    print_value = await bidi_session.browsing_context.print(
+
+    # This yields and empty content area: https://github.com/w3c/webdriver-bidi/issues/473
+    with pytest.raises(UnsupportedOperationException):
+        await bidi_session.browsing_context.print(
+            context=top_context["context"],
+            shrink_to_fit=False,
+            margin=margin,
+        )
+
+
+@pytest.mark.parametrize(
+    "margin",
+    [
+        {"top": 27.94 - ceil(inch_in_cm / inch_in_point)},
+        {"left": 21.59 - ceil(inch_in_cm / inch_in_point)},
+        {"right": 21.59 - ceil(inch_in_cm / inch_in_point)},
+        {"bottom": 27.94 - ceil(inch_in_cm / inch_in_point)},
+    ],
+    ids=[
+        "top",
+        "left",
+        "right",
+        "bottom",
+    ],
+)
+async def test_minimum_page_size_with_individual_margins(
+    bidi_session,
+    top_context,
+    inline,
+    assert_pdf_content,
+    assert_pdf_dimensions,
+    margin,
+):
+    page = inline("Text")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=page, wait="complete"
+    )
+
+    value = await bidi_session.browsing_context.print(
         context=top_context["context"],
         shrink_to_fit=False,
-        margin=margin,
+        margin=margin
     )
 
-    # Check that content is out of page dimensions.
-    await assert_pdf_content(print_value, [{"type": "string", "value": ""}])
+    # Check that margins don't affect page dimensions and equal in this case defaults.
+    await assert_pdf_dimensions(value, {
+       "width": 21.59 - (0.0 if ("top" in margin or "bottom" in margin) else inch_in_cm / inch_in_point),
+       "height": 27.94 - (0.0 if ("left" in margin or "right" in margin) else inch_in_cm / inch_in_point),
```

Comment:
Lets move the calculation out to `expected_width` and `expected_height` and pass that in instead.

Reply:
Done